tools/bake.py

```diff
@@ -367,6 +368,36 @@ def bake_pass(self, context, bake_name, bake_type, bake_pass_filter, objects, ba
                     for node in obj.active_material.node_tree.nodes:
                         if node.type == "VALUE" and node.label == "bake_" + bake_name:
                             node.outputs["Value"].default_value = 0
+        
+        
+        if context.scene.bake_optimize_solid_materials: 
+            #arranging old pixels and assignment to image pixels this way makes only one update per pass, so many many times faster - @989onan
+            old_pixels = image.pixels[:]
+            
+            #lastly, slap our solid squares on top of bake atlas, to make a nice solid square without interuptions from the rest of the bake - @989onan
+            for child in [obj for obj in objects if obj.type == "MESH"]: #grab all mesh objects being baked
+                for matindex,material in enumerate(child.data.materials):
+                    if material.name in solidmaterialcolors:
+                        index = list(solidmaterialcolors.keys()).index(material.name)
+                        old_pixels = list(old_pixels)
+                        #in pixels
+                        square_center_coord = (((-(((((index*2)+1)%(bake_size[0]/4))*4)-(bake_size[0]/2))/bake_size[0])*2)*bake_size[0],   (((((int(((index*2)+1)/(bake_size[1]/4))*-8)+((bake_size[1]/2)-4)))/bake_size[1])*2)*bake_size[1])
```

Comment:
This isn't really readable...

Reply:
It's an equation to wrap, and I have no idea how to write it better... It's supposed to wrap around the image for the X and Y cords. Do you think there's a better way to write this?

Reply:
I got help from @Sacred#9619 on discord. Put him in code comment for credit on that one